third_party/terraform/resources/resource_compute_instance.go.erb

```diff
@@ -1388,50 +1421,17 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 			}
 		}
 
-		if d.HasChange(prefix + ".access_config") {
-
-			// TODO: This code deletes then recreates accessConfigs.  This is bad because it may
-			// leave the machine inaccessible from either ip if the creation part fails (network
-			// timeout etc).  However right now there is a GCE limit of 1 accessConfig so it is
-			// the only way to do it.  In future this should be revised to only change what is
-			// necessary, and also add before removing.
-
-			// Delete any accessConfig that currently exists in instNetworkInterface
-			for _, ac := range instNetworkInterface.AccessConfigs {
-				op, err := config.NewComputeClient(userAgent).Instances.DeleteAccessConfig(
-					project, zone, instance.Name, ac.Name, networkName).Do()
-				if err != nil {
-					return fmt.Errorf("Error deleting old access_config: %s", err)
-				}
-				opErr := computeOperationWaitTime(config, op, project, "old access_config to delete", userAgent, d.Timeout(schema.TimeoutUpdate))
-				if opErr != nil {
-					return opErr
-				}
+		if !updateDuringStop && d.HasChange(prefix+".access_config") {
+			// Delete current access configs
```

Comment:
To clarify- *this* is where the comment about deleting and then creating should go, because this is the code that controls that order.

Reply:
oof, ooops thanks for the heads up.